### PR TITLE
Interfaced Yukawa Libint2 ERI Kernel

### PR DESCRIPTION
**Pull request by andyj10224**
## Description
Per issue #2374, this PR adds an interface to Yukawa ERIs from Libint2.

## Questions
- [x] How would I test my new code? (How are erf kernels tested?)
- [x] Do I need to add functions in MintsHelper for the Yukawa kernel?

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by susilehtola** [comment]
I'm not sure if there are any Gaussian codes that have the Yukawa kernels. However, I do have the Yukawa kernels in HelFEM for atomic calculations, and for `hyb_gga_xc_camy_b3lyp` I get the following energies

| atom | energy |
|--------|-----------|
| He    |     -2.90309239 |
| N      |   -54.59195697 |
| Ne    | -128.96948058 |

If you run the calculations in Psi4 with the AHGBS-9 basis (available from the BSE) and a large quadrature grid like (250, 590), you should be able to match these to sub-microhartree accuracy.

**Comment by susilehtola** [comment]
Of course, to run those tests one also needs to add the new kernels to the range-separated Fock builds and the libxc interface.

File: psi4/src/psi4/libmints/integral.h
Comment (JonathonMisiewicz) [inline psi4/src/psi4/libmints/integral.h:503]: Returning a raw pointer is dangerous. Can we possibly make this a `unique_ptr`, or will that create issues elsewhere down the line?

**Comment by JonathonMisiewicz** [comment]
I'll wait for reviews from @jturney and @andysim before commenting further. I don't know this part of the codebase well.

**Comment by andyj10224** [comment]
> Of course, to run those tests one also needs to add the new kernels to the range-separated Fock builds and the libxc interface.

I believe that is outside the scope of this PR, is there I way I can test for numerical correctness? (Possibly expose the functions to python and call a numpy function)?

File: psi4/src/psi4/libmints/integral.cc
Comment (jturney) [inline psi4/src/psi4/libmints/integral.cc:263]: Modify the function to only use libint2 since no other engine supports it.

File: psi4/src/psi4/libmints/integral.h
Comment (jturney) [reply psi4/src/psi4/libmints/integral.h:503]: Leaving it as a raw pointer follows the pattern of the other integrals. Another PR could be done that makes the conversion to a unique_ptr.

**Review by jturney — CHANGES_REQUESTED** [review]

File: psi4/src/psi4/libmints/integral.cc
Comment (andyj10224) [reply psi4/src/psi4/libmints/integral.cc:263]: I currently have the function throw an exception if the integral package is not set to Libint2. Should I just let the Yukawa ERI be created regardless of what the user option is set as?

Comment (jturney) [reply psi4/src/psi4/libmints/integral.cc:263]: Yes. We will always require libint2 to be installed. If another package is added that supports this integral then this ability can be added back.

Comment (andyj10224) [reply psi4/src/psi4/libmints/integral.cc]: Fixed this

**Review by jturney — APPROVED** [review]

**Review by JonathonMisiewicz — APPROVED** [review]
As far as I'm aware, the only other issue is testing, and we don't have a good way to test without massively increasing the scope of the PR. As long as the integrals aren't _exposed_ until we're sure they work, they should be fine. Anybody who would be using the Yukawa integrals assumes the risk.

File: psi4/src/psi4/libmints/eri.cc
Comment (loriab) [inline psi4/src/psi4/libmints/eri.cc:521]:
```suggestion
            throw PSIEXCEPTION("Bad BraKet type in Libint2YukawaERI::libint2_wrapper0");
```

File: psi4/src/psi4/libmints/eri.cc
Comment (loriab) [inline psi4/src/psi4/libmints/eri.cc:541]:
```suggestion
            throw PSIEXCEPTION("Bad BraKet type in Libint2YukawaERI::libint2_wrapper1");
```

File: psi4/src/psi4/libmints/eri.cc
Comment (loriab) [inline psi4/src/psi4/libmints/eri.cc:560]:
```suggestion
            throw PSIEXCEPTION("Bad BraKet type in Libint2YukawaERI::libint2_wrapper2");
```

File: psi4/src/psi4/libmints/eri.cc
Comment (loriab) [inline psi4/src/psi4/libmints/eri.cc:491]:
```suggestion
        throw PSIEXCEPTION("Bad BraKet type in Libint2YukawaERI");
```
questionable, as preceding two-electron ints types do this, too

**Review by loriab — APPROVED** [review]
lgtm!

**Pull request merged**
